server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfilesWs.java

```diff
@@ -28,13 +28,16 @@
   private final QProfileRestoreBuiltInAction resetAction;
   private final RuleActivationActions ruleActivationActions;
   private final BulkRuleActivationActions bulkRuleActivationActions;
+  private final ProjectAssociationActions projectAssociationActions;
 
   public QProfilesWs(QProfileRestoreBuiltInAction resetAction,
                      RuleActivationActions ruleActivationActions,
-                     BulkRuleActivationActions bulkRuleActivationActions) {
+                     BulkRuleActivationActions bulkRuleActivationActions,
+                     ProjectAssociationActions projectAssociationActions) {
```

Comment:
I think it should be better to use an interface for these actions, a kind of BaseQProfileAction (see BaseIssuesWsAction for instance)